addons/messaging-executor/impl/src/main/java/org/jboss/windup/web/messaging/executor/HttpPostSerializer.java

```diff
@@ -0,0 +1,190 @@
+            throw new RuntimeException("No '" + SYSTEM_PROPERTY_HTTP_DOWNLOAD_URL + "' System Property specified!");
+
+        url = url + "/" + request.getExecution().getId();
+
+        try (InputStream httpIS = new URL(url).openStream())
+        {
+            TarInputStream tarInputStream = new TarInputStream(httpIS);
+            TarEntry tarEntry;
+            while ((tarEntry = tarInputStream.getNextEntry()) != null)
+            {
+                int count;
+                byte data[] = new byte[32768];
+                String outputFile = inputFilesDirectory + "/" + tarEntry.getName();
+                newInputFiles.add(outputFile);
+                LOG.info("Creating output file: " + outputFile);
+                FileOutputStream fos = new FileOutputStream(outputFile);
+                try (BufferedOutputStream dest = new BufferedOutputStream(fos))
+                {
+                    while ((count = tarInputStream.read(data)) != -1)
+                    {
+                        dest.write(data, 0, count);
+                    }
+                }
+            }
+        }
+
+        request.getExecution().getAnalysisContext().getApplications()
+                    .forEach(application -> {
+                        for (String newInputFile : newInputFiles)
+                        {
+                            if (Paths.get(newInputFile).getFileName().toString()
+                                        .equals(Paths.get(application.getInputPath()).getFileName().toString()))
+                            {
+                                application.setInputPath(newInputFile);
+                            }
+                        }
+                    });
+    }
+
+    private void sendResults(long executionId, InputStream inputStream) throws IOException
+    {
+        String url = System.getProperty(SYSTEM_PROPERTY_HTTP_POST_URL);
+        LOG.info("Posting results to: " + url);
+        if (StringUtils.isBlank(url))
+            throw new RuntimeException("No '" + SYSTEM_PROPERTY_HTTP_POST_URL + "' System Property specified!");
+
+        url = url + "/" + executionId;
+
+        HttpClient httpClient = HttpClientBuilder.create().build();
+        HttpContext httpContext = new BasicHttpContext();
+
+        String fileName = "execution_files.tar";
+
+        HttpPost httpPost = new HttpPost(url);
+        HttpEntity entity = MultipartEntityBuilder.create()
+                    .addPart("file", new InputStreamBody(inputStream, ContentType.APPLICATION_OCTET_STREAM, fileName))
+                    .build();
+        httpPost.setEntity(entity);
+        HttpResponse postResponse = httpClient.execute(httpPost, httpContext);
+        int responseCode = postResponse.getStatusLine().getStatusCode();
+        if (responseCode < 200 || responseCode >= 300)
+        {
+            throw new RuntimeException("Failed to post results due to error: " + postResponse.getStatusLine().getReasonPhrase());
+        }
```

Comment:
It would be great to also have a log entry that pairs w/ the one above "Posting results to: " so that user can see in logs when upload has completed